packages/@sanity/desk-tool/src/panes/document/statusBar/sparkline/DocumentSparkline.tsx

```diff
@@ -1,74 +1,85 @@
+    value,
+    documentId,
+    documentType,
+    handleHistoryClose,
+    handleHistoryOpen,
+    changesOpen,
+  } = useDocumentPane()
+  const syncState = useSyncState(documentId, documentType)
+
   const lastUpdated = value?._updatedAt
   const lastPublished = editState?.published?._updatedAt
   const showingRevision = historyController.onOlderRevision()
   const liveEdit = Boolean(editState?.liveEdit)
   const published = Boolean(editState?.published)
   const changed = Boolean(editState?.draft)
-  const loaded = published || changed
 
-  // Keep track of the size of the review changes button
-  const [
-    reviewChangesButtonElement,
-    setReviewChangesButtonElement,
-  ] = useState<HTMLButtonElement | null>(null)
-  const reviewChangesButtonElementRect = useElementRect(reviewChangesButtonElement)
-  const reviewChangesButtonWidth = reviewChangesButtonElementRect?.width
-  const reviewChangesButtonWidthRef = useRef(reviewChangesButtonWidth || 0)
+  const [rootFlexElement, setRootFlexElement] = useState<HTMLDivElement | null>(null)
+  const rootFlexRect = useElementRect(rootFlexElement)
+  const collapsed = Boolean(rootFlexRect && rootFlexRect?.width < 300)
 
-  // Use the size of the review changes button as a element query breakpoint
-  const metadataBoxBreakpoints = useMemo(
-    () => [reviewChangesButtonWidth || reviewChangesButtonWidthRef.current, 225],
-    [reviewChangesButtonWidth]
-  )
+  const [status, setStatus] = useState<'saved' | 'syncing' | null>(null)
 
-  // Only transition between subsequent state, not the initial
-  const [transition, setTransition] = useState(false)
+  // eslint-disable-next-line consistent-return
   useEffect(() => {
-    if (!transition && loaded) {
-      // NOTE: the reason for double RAF here is a common "bug" in browsers.
-      // See: https://stackoverflow.com/questions/44145740/how-does-double-requestanimationframe-work
-      // There is no need to cancel this animation,
-      // since calling it again will cause the same result (transition=true).
-      return raf2(() => setTransition(true))
+    if (status === 'syncing') {
+      // status changed to 'syncing', schedule an update to set it to 'saved'
+      const timerId = setTimeout(() => setStatus('saved'), 1000)
+      return () => clearTimeout(timerId)
     }
+    if (status === 'saved') {
+      // status changed to 'saved', schedule an update to clear it
+      const timerId = setTimeout(() => setStatus(null), 3000)
+      return () => clearTimeout(timerId)
+    }
+  }, [status, lastUpdated])
 
-    return undefined
-  }, [loaded, transition])
-
-  const metadataBoxStyle = useMemo(
-    () =>
-      ({
-        '--session-layout-width': reviewChangesButtonWidth
-          ? `${reviewChangesButtonWidth}px`
-          : undefined,
-      } as React.CSSProperties),
-    [reviewChangesButtonWidth]
-  )
+  useLayoutEffect(() => {
+    // clear sync status when documentId changes
+    setStatus(null)
+  }, [documentId])
```

Comment:
Should this hook also clear the timers? E.g.:

```
clearTimeout(timerId)
```

Reply:
Not sure if that is needed?